Index: wysiwyg.module
===================================================================
--- wysiwyg.module	(revision 3707)
+++ wysiwyg.module	(working copy)
@@ -170,6 +170,10 @@
               // Check editor theme (and reset it if not/no longer available).
               $theme = wysiwyg_get_editor_themes($profile, (isset($profile->settings['theme']) ? $profile->settings['theme'] : ''));
 
+              // Rough estimate of textarea height in pixels -lwalley
+              // 40 for the toolbar, 29 for the first row, and 15 for the rest.
+              $profile->settings['height'] = $field['#rows'] ? 40 + 29 + 15*($field['#rows'] - 1) : 400;
+
               // Add plugin settings (first) for this input format.
               wysiwyg_add_plugin_settings($profile);
               // Add profile settings for this input format.
@@ -254,6 +258,7 @@
     if ($editor) {
       // Determine library files to load.
       // @todo Allow to configure the library/execMode to use.
+      // For edulifedesks this means the height will be taken from the row value of the first field of each format type (e.g. field_spmi_biology in taxon_page) and applied to all fields of that type in any node form -lwalley.
       if (isset($profile->settings['library']) && isset($editor['libraries'][$profile->settings['library']])) {
         $library = $profile->settings['library'];
         $files = $editor['libraries'][$library]['files'];
Index: editors/tinymce.inc
===================================================================
--- editors/tinymce.inc	(revision 3707)
+++ editors/tinymce.inc	(working copy)
@@ -146,6 +146,7 @@
     'plugins' => array(),
     'theme' => $theme,
     'width' => '100%',
+    'height' => $config['height'] ? $config['height'] : '400',
     // Strict loading mode must be enabled; otherwise TinyMCE would use
     // document.write() in IE and Chrome.
     'strict_loading_mode' => TRUE,